core/network.rb

```diff
@@ -177,62 +177,83 @@ def self.show(name)
+          result_ip.push(getNodeParam('IP',node, box_params))
         end
       else
-        mdbci_node = $session.mdbciNodes.find { |elem| elem[0].to_s == args[1] }
-        raise "MDBCI node #{args[1]} is not found in #{args[0]}" if mdbci_node.nil?
-        box = mdbci_node[1]['box'].to_s
-        raise "Can not find box parameter for node #{args[1]}" if !box.empty?
-        mdbci_params = $session.boxes.getBox(box)
-        raise "Can not find box #{box} node #{args[1]} in #{args[0]}" if !box.empty?
-        $out.info 'Node: ' + args[1].to_s
-        $out.out mdbci_params['IP'].to_s
-
+        mdbci_node = $session.mdbciNodes.find { |elem| elem[0].to_s == node_arg }
+        raise "MDBCI node #{node_arg} is not found in #{dir}" if mdbci_node.nil?
+        box_params = getBoxParams(node[1])
+        result_ip.push(getNodeParam('IP',mdbci_node, box_params))
       end
     else # aws, vbox nodes
-      raise "Can not find directory #{args[0]}" unless Dir.exists? args[0]
+      raise "Can not find directory #{dir}" unless Dir.exists? dir
       network = Network.new
-      network.loadNodes pwd.to_s+'/'+args[0] # load nodes from dir
-      if args[1].nil? # No node argument, show all config
-        raise "Nodes are not found in #{args[0]}" if network.nodes.empty?
+      network.loadNodes pwd.to_s+'/'+dir # load nodes from dir
+      raise "Nodes are not found in #{dir}" if network.nodes.empty?
+      if node_arg.nil? # No node argument, show all config
         network.nodes.each do |node|
-          exit_code = node.getIp(node.provider, true)
-          raise "Can not get IP for #{node.name} in #{args[0]}" if exit_code != 0
-          $out.info 'Node: ' + node.name
-          $out.out node.ip.to_s
+          result_ip.push(getNodeIP(node))
         end
       else
-        node = network.nodes.find { |elem| elem.name == args[1]}
-        raise "Node #{args[1]} is not found in #{args[0]}" if node.nil?
-        exit_code = node.getIp(node.provider, true)
-        raise "Can not get IP for #{node.name} in #{args[0]}" if exit_code != 0
-        $out.info 'Node: ' + node.name
-        $out.out node.ip.to_s
+        node = network.nodes.find { |elem| elem.name == node_arg}
+        raise "Node #{node_arg} is not found in #{dir}" if node.nil?
+        result_ip.push(getNodeIP(node))
       end
     end
     Dir.chdir pwd
+    return result_ip
+  end
 
-    return 0
+  def self.getNodeIP(node)
+    result = Hash.new("")
+    exit_code = node.getIp(node.provider, true)
+    raise "Can not get IP for #{node.name} in #{dir}" if exit_code != 0
+    result["Node"] = node.name.to_s
+    result["IP"] = node.ip.to_s
+    return result
+  end
+
+  def self.getNodeParam(param,node_name,box_params)
+    result = Hash.new("")
+    result["Node"] = node_name.to_s
+    result[param.to_s] = box_params[param].to_s
+    return result    
+  end
+
+  def self.showArrayHash(array,params)
+    array.each do |hash|
+      params.each do |param|
```

Comment:
Please refactor this code in a following manner:
IPs should be printed without any additional strings to $out.out
Node names should be printed without any additional strings to $out.info